core/src/main/java/com/netflix/conductor/validations/TaskDefConstraint.java

```diff
@@ -0,0 +1,25 @@
+package com.netflix.conductor.validations;
+
+import com.google.inject.Singleton;
+import com.google.inject.multibindings.ProvidesIntoSet;
+import com.netflix.conductor.common.metadata.workflow.WorkflowDef;
+import com.netflix.conductor.common.metadata.workflow.WorkflowTask;
+import org.hibernate.validator.HibernateValidatorConfiguration;
+import org.hibernate.validator.cfg.ConstraintMapping;
+
+import static java.lang.annotation.ElementType.FIELD;
+
+public class TaskDefConstraint {
```

Comment:
As mentioned earlier, is this really needed?

Reply:
This is required because it needs access to DAO to validate workflow Task name. Hence programmatic constraints are defined.